tests/TestContextHttpTests.elm

```diff
@@ -3,100 +3,265 @@ module TestContextHttpTests exposing (all)
 import Expect exposing (Expectation)
 import Html
 import Html.Events exposing (onClick)
+import Json.Decode
+import SimulatedEffect.Http as Http
+import SimulatedEffect.Task as Task
 import Test exposing (..)
+import Test.Http
 import Test.Runner
 import TestContext exposing (TestContext)
 
 
-type TestEffect
-    = NoEffect
-    | HttpGet String
+type alias TestEffect =
+    List (TestContext.SimulatedEffect TestMsg)
 
 
-deconstructEffect : TestEffect -> List TestContext.SimulatedEffect
-deconstructEffect testEffect =
-    case testEffect of
-        NoEffect ->
-            []
+type TestMsg
+    = PassThroughEffect TestEffect
+    | HandleFriendsResponse (Result Http.Error (List String))
+    | HandleStringResponse (Result Http.Error String)
 
-        HttpGet url ->
-            [ TestContext.HttpRequest { method = "GET", url = url } ]
 
+type alias TestModel =
+    String
 
-type alias TestMsg =
-    TestEffect
 
-
-start : TestEffect -> TestContext TestMsg () TestEffect
+start : TestEffect -> TestContext TestMsg TestModel TestEffect
 start initialEffect =
     TestContext.createElement
-        { init = \() -> ( (), initialEffect )
-        , update = \msg () -> ( (), msg )
+        { init = \() -> ( "Init", initialEffect )
+        , update =
+            \msg model ->
+                case msg of
+                    PassThroughEffect effect ->
+                        ( model, effect )
+
+                    HandleFriendsResponse result ->
+                        ( Debug.toString result, [] )
+
+                    HandleStringResponse result ->
+                        ( Debug.toString result, [] )
         , view =
-            \() ->
+            \_ ->
                 Html.div []
                     [ Html.button
-                        [ onClick (HttpGet "https://example.com/buttons/get") ]
+                        [ onClick
+                            (PassThroughEffect
+                                [ Http.get
+                                    { url = "https://example.com/buttons/get"
+                                    , expect =
+                                        Http.expectJson HandleFriendsResponse
+                                            (Json.Decode.list Json.Decode.string)
```

Comment:
Is this assuming that I, as the consumer of this library, do my data manipulation in my view before calling my update?

Reply:
This is a weird thing for this test app -- in the real world, your onClick would be a Msg like `SubmitForm`, and then your update function would be producing an `effect` that would become this list of SimulatedEffect in the test.